### DISCUSSION TITLE
How to generate TurbSIM.exe?

### DISCUSSION THREAD

**Discussion by elalond3** [OpenFAST/openfast#827] [Q&A] [answered]
Apologies if this is the wrong spot to ask for this, but I've been using FAST v8 for many years but only recently swapped to OpenFAST, mostly because I struggle with GitHub. I figured out to how download OpenFAST, but I haven't been able to find/create the equivalent version of TurbSim, and it seems that the older version 1.06 is incompatible with OpenFAST. The documentation says that you need to "make" Turbsim using visual studio or something like that, but I am totally lost on how to do that. Is there a video or step-by-step guide that shows this process in more detail than the documentation? Thanks

**Comment by jjonkman** [comment, marked as answer]
Dear @elalond3,

I'm not aware of incompatibilities between TurbSim v1.06 (although I have not used TurbSim v.1.06 for many years, so, perhaps there is something I forgot); can you clarify what issue you have when using TurbSim v1.06 with FAST v8 or OpenFAST?

Between TurbSim v1.06 and the version of TurbSim developed within OpenFAST, we had released TurbSim v2, which you can find here: https://www.nrel.gov/wind/nwtc/assets/downloads/TurbSim/TurbSim_v2.00.07a-bjj.exe.  The version of TurbSim developed within OpenFAST is based on TurbSim v2 with only a few modifications to date.

Regarding compiling TurbSim, we don't have a video or a step-by-step guide, but the process should be very similar to the compilation of OpenFAST.  On Windows, in my opinion the use of Visual Studio with Intel Fortran is the most straightforward, in which case installing Visual Studio with Intel Fortran is far harder than actually compiling.  Once installed, all you need to do is open _TurbSim.vfproj_ and click on Build.  On mac or linux, the use of CMake to compile is likely preferred.

Best regards,

**Reply by elalond3** [reply]
Dr. Jonkman,

Thanks for the reply. I think I have been mistaken about the incompatibility between TurbSim and OpenFAST - I tried it with a .bts generated using v2.0 and got the same error. I was running into an error with the FF wind array boundaries, and after reading an exchange on a similar error (https://wind.nrel.gov/forum/wind/viewtopic.php?t=1450) I assumed it was a compatibility issue based on the final post there. It seems I might be dealing with another issue and TurbSim was working fine all along. Still - thank you for the link, I'm glad to update. It may be worth considering to include that link in Section 4.11.2 of the doc for those just looking to run so basic sims with OpenFAST and TurbSim
